dozer/sources/TwitchSource.py

```diff
@@ -0,0 +1,208 @@
+        client_id = self.bot.config['news']['twitch']['client_id']
+        self.client_id = client_id
+        client_secret = self.bot.config['news']['twitch']['client_secret']
+        params = {
+            'client_id': client_id,
+            'client_secret': client_secret,
+            'grant_type': 'client_credentials'
+        }
+        response = await self.http_session.post(self.token_url, params=params)
+        response = await response.json()
+        try:
+            self.access_token = response['access_token']
+        except KeyError:
+            DOZER_LOGGER.critical(f"Error in {self.full_name} Token Get: {response['message']}")
+            self.disabled = True
+            return
+
+        expiry_seconds = response['expires_in']
+        time_delta = datetime.timedelta(seconds=expiry_seconds)
+        self.expiry_time = datetime.datetime.now() + time_delta
+
+    async def request(self, url, *args, headers=None, **kwargs):
+        """Make a OAuth2 verified request to a API Endpoint"""
+        if headers is None:
+            headers = {'Authorization': f"Bearer {self.access_token}",
+                       "Client-ID": self.client_id}
+        else:
+            headers['Authorization'] = f"Bearer {self.access_token}"
+
+        url = f"{self.api_url}/{url}"
+
+        response = await self.http_session.get(url, headers=headers, *args, **kwargs)
+
+        if response.status == 401:
+            if 'WWW-Authenticate' in response.headers:
+                DOZER_LOGGER.info("Twitch token expired when request made, request new token and retrying.")
+                await self.get_token()
+                return await self.request(url, headers, *args, **kwargs)
+
+        json = await response.json()
+        return json
+
+    async def first_run(self, data=None):
+        """Make sure we have a token, then verify and add all the current users in the DB"""
+        await self.get_token()
+
+        if not data:
+            return
+
+        params = []
+        for login in data:
+            params.append(('login', login))
+        json = await self.request("users", params=params)
+        for user in json['data']:
+            user_obj = TwitchSource.TwitchUser(user['id'], user['display_name'], user['profile_image_url'],
+                                               user['login'])
+            self.users[user['id']] = user_obj
+
+    async def clean_data(self, text):
+        """Request user data from Twitch to verify the username exists and clean the data"""
+        try:
+            user_obj = self.users[text]
+        except KeyError:
+            json = await self.request('users', params={'login': text})
+            if len(json['data']) == 0:
+                raise DataBasedSource.InvalidDataException("No user with that login name found")
+            elif len(json['data']) > 1:
+                raise DataBasedSource.InvalidDataException("More than one user with that login name found")
```

Comment:
In this case, what can the user do to indicate which streamer they want?

Reply:
I have no clue. I have no idea if this is even possible. I added error handling just in case.